### PR TITLE
N°5950 - cleanup code and fix some futurs bugs in php 8.2

### PR DESCRIPTION
**Pull request by accognet**
small improvements :
-add missing variable declarations
-remove code deprecated in php 8.1 and useless (used only for old Mac system)
-remove test on JSON_PRETTY_PRINT (defined since php 5.3) 
-fix a comment

### CODE REVIEW

**Comment by Molkobain** [comment]
Hello Anne-Cath, please create the corresponding ticket in the bug tracker and detail all those changes please as they will be copied to the collectors changelogs.

File: core/collector.class.inc.php [hunk 1]
```diff
@@ -298,8 +299,10 @@ protected function DoProcessBeforeSynchro()
 			if (preg_match($sPattern, $sDataFile, $aMatches)) {
 				$idx = $aMatches[1];
 				$sOutputFile = Utils::GetDataFilePath(get_class($this).'-'.$idx.'.csv');
+
 				Utils::Log(LOG_DEBUG, "Converting '$sDataFile' to '$sOutputFile'...");
-				ini_set('auto_detect_line_endings', true);
+				//only usefull of files from old mac system deprecated in php 8.1
+				//ini_set('auto_detect_line_endings', true);
```
Comment (Molkobain) [inline core/collector.class.inc.php:305]: Completely remove these lines then please.

**Comment by Hipska** [comment]
JSON_PRETTY_PRINT is only available [since 5.4](https://3v4l.org/PJLdJ), see also discussion on this in #10.

File: core/collector.class.inc.php (hunk 1 above)
Comment (Molkobain) [reply core/collector.class.inc.php:305]: Add unit test files with Unix / Windows / Classic MacOS line endings to ensure that this still works as excepted.

Comment (Molkobain) [reply core/collector.class.inc.php:305]: @dflaven does this line ring any bell?

**Review by Molkobain — CHANGES_REQUESTED** [review]
See comments

File: core/collector.class.inc.php (hunk 1 above)
Comment (Molkobain) [reply core/collector.class.inc.php:305]: Discussed with Anne-Cath, this setting is only used by `fgets()` and `file()` which are not used in the collectors.

**Review by Molkobain — APPROVED** [review]

File: core/collector.class.inc.php (hunk 1 above)
Comment (Hipska) [reply core/collector.class.inc.php:305]: The comments and commented code still needs to be removed? No?

Comment (accognet) [reply core/collector.class.inc.php]: Yes it's done

**Pull request merged**